MaeYoungPhan/md-bay-game-server
# initial records of trips view complete along with urls

# Description of Proposed Changes
List, Retrieve, Create, Update, Destroy added to record of trips view.
List shows only current user's trips
Retrieve has option to i.d. current user's trips using author property decorator that was added to ROT model

---


# Steps to Test

[ ] Fetch branch and switch to it, run debugger
[ ] Open Postman
[ ] GET request to http://localhost:8000/recordoftrips should return list of only current user's trips
[ ] PUT request to http://localhost:8000/recordoftrips/12 RESPONSE BODY: None Status: 204 verify appropriate trip was updated in GET request 
Sample input for gamer 6, dictionary 12:
{
            "gamer": 6,
            "date": "2023-07-14",
            "name": "Home",
            "number_found": 11,
            "occasion": 13
    }
[ ]  POST request to http://localhost:8000/recordoftrips RESPONSE BODY: JSON object Status: 201 
Sample input for gamer 6:
{
            "gamer": 6,
            "date": "2023-07-14",
            "name": "Party",
            "number_found": 11,
            "occasion": 9
    }
[ ] GET request to http://localhost:8000/recordoftrips/{postyoujustcreated} RESPONSE BODY: JSON object Status: 200
[ ] DELETE request to http://localhost:8000/recordoftrips/{postyoujustcreated} RESPONSE BODY: None Status: 204
[ ] GET request to http://localhost:8000/recordoftrips/{postyoujustdeleted} RESPONSE BODY: None Status: 404
[ ] Proofread files changed
